security/security-spring/src/main/java/org/cbioportal/security/spring/authentication/keycloak/SAMLUserDetailsServiceImpl.java

```diff
@@ -0,0 +1,142 @@
+{
+    private static final Log log = LogFactory.getLog(SAMLUserDetailsServiceImpl.class);
+
+    @Value("${saml.idp.metadata.attribute.email:}")
+    private String SAML_IDP_METADATA_EMAIL_ATTR_NAME;
+
+    @Value("${saml.idp.metadata.attribute.role:}")
+    private String SAML_IDP_METADATA_ROLE_ATTR_NAME;
+
+    @Value("${app.name:}")
+    private String APP_NAME;
+    
+
+    /**
+     * Default no_arg Constructor.
+     */
+    public SAMLUserDetailsServiceImpl() {
+    }
+
+    /**
+     * 
+     * @param attributeValue
+     * @return String type attribute value
+     */
+    private String getAttributeValue(XMLObject attributeValue)
+    {
+        return attributeValue == null ?
+            null :
+            attributeValue instanceof XSString ?
+                ((XSString) attributeValue).getValue() :
+                attributeValue instanceof XSAnyImpl ?
+                    ((XSAnyImpl) attributeValue).getTextContent() :
+                    attributeValue.toString();
+    }
+
+    /**
+     * Implementation of {@code SAMLUserDetailsService}. 
+     * Instantiate PortalUserDetails object from  
+     * SAML assertion received from keycloak Identity Provider
+     */
+    @Override
+    public Object loadUserBySAML(SAMLCredential credential)
+    {
+        PortalUserDetails userDetailsObj = null;
+        String userId = null;
+        List<String> userRoles = new ArrayList<String>();
+
+        // get userid and roles: iterate over attributes searching for "email" and "roles":
+        for (Attribute cAttribute : credential.getAttributes()) {
+            String attrName = cAttribute.getName();
+            log.debug("loadUserBySAML(), parsing attribute -" + attrName);
+
+
+            if (userId == null && attrName.equals(SAML_IDP_METADATA_EMAIL_ATTR_NAME)) {
+                userId = credential.getAttributeAsString(cAttribute.getName());
+                log.debug(userId);
+            }
+            if (attrName.equals(SAML_IDP_METADATA_ROLE_ATTR_NAME)) {
+                List<XMLObject> attributeValues = cAttribute.getAttributeValues();
+                if (!attributeValues.isEmpty()) {
+                    userRoles.add(new StringBuilder(APP_NAME).append(":").append(
+                        getAttributeValue(attributeValues.get(0))).toString());
+                }    
+            }
+        }
+        
+        try {
+            
+            if (userId == null) {
+
+                String errorMsg = "loadUserBySAML(), can not instantiate PortalUserDetails from SAML assertion." 
+                    + "Expected 'email' attribute was not found or has no values. ";
+                log.error(errorMsg);
+                throw new Exception(errorMsg);
+            }
+
+            log.debug("loadUserBySAML(), IDP successfully authenticated user, userid: " + userId);
+
+            //add granted authorities:
+            if (userRoles.size() > 0) userDetailsObj = new PortalUserDetails(userId,
```

Comment:
Google style dictates to always use braces even with a single statement. I think it makes the code more clear, but it's not necessary.